internal/worker/bootstrap/worker.go

```diff
@@ -479,7 +483,14 @@ func (w *bootstrapWorker) seedAgentBinary(ctx context.Context, dataDir string) (
 
 	// Agent binary seeder will populate the tools for the agent.
 	agentStorage := agentStorageShim{State: w.cfg.SystemState}
-	cleanup, err := w.cfg.AgentBinaryUploader(ctx, dataDir, agentStorage, objectStore, w.cfg.Logger.Child("agentbinary"))
+	cleanup, err := w.cfg.AgentBinaryUploader(
+		ctx,
+		dataDir,
+		agentStorage,
```

Comment:
`agentStorage` is the legacy mongo store right?
Maybe a brief comment to say that?

Reply:
Good point